genesyscloud/routing_queue_conditional_group_routing/genesyscloud_routing_queue_conditional_group_routing_proxy.go

```diff
@@ -0,0 +1,172 @@
+	// Get the routing queue the rules belong to
+	queue, resp, err := p.routingApi.GetRoutingQueue(queueId)
+	if err != nil {
+		return nil, resp, fmt.Errorf("routing queue %s not found: %s", queueId, err)
+	}
+
+	groupRoutingObj := platformclientv2.Conditionalgrouprouting{Rules: rules}
+
+	// Copy over all the values from the original object to the new object
+	updateQueue := platformclientv2.Queuerequest{
+		Name:                         queue.Name,
+		Description:                  queue.Description,
+		MemberCount:                  queue.MemberCount,
+		UserMemberCount:              queue.UserMemberCount,
+		JoinedMemberCount:            queue.JoinedMemberCount,
+		MediaSettings:                queue.MediaSettings,
+		RoutingRules:                 queue.RoutingRules,
+		ConditionalGroupRouting:      &groupRoutingObj, // Add the new rules
+		Bullseye:                     queue.Bullseye,
+		ScoringMethod:                queue.ScoringMethod,
+		AcwSettings:                  queue.AcwSettings,
+		SkillEvaluationMethod:        queue.SkillEvaluationMethod,
+		MemberGroups:                 queue.MemberGroups,
+		QueueFlow:                    queue.QueueFlow,
+		EmailInQueueFlow:             queue.EmailInQueueFlow,
+		MessageInQueueFlow:           queue.MessageInQueueFlow,
+		WhisperPrompt:                queue.WhisperPrompt,
+		OnHoldPrompt:                 queue.OnHoldPrompt,
+		AutoAnswerOnly:               queue.AutoAnswerOnly,
+		EnableTranscription:          queue.EnableTranscription,
+		EnableAudioMonitoring:        queue.EnableAudioMonitoring,
+		EnableManualAssignment:       queue.EnableManualAssignment,
+		AgentOwnedRouting:            queue.AgentOwnedRouting,
+		DirectRouting:                queue.DirectRouting,
+		CallingPartyName:             queue.CallingPartyName,
+		CallingPartyNumber:           queue.CallingPartyNumber,
+		DefaultScripts:               queue.DefaultScripts,
+		OutboundMessagingAddresses:   queue.OutboundMessagingAddresses,
+		PeerId:                       queue.PeerId,
+		SuppressInQueueCallRecording: queue.SuppressInQueueCallRecording,
+	}
+
+	// For some reason OutboundEmailAddress returned by GetRoutingQueue is a pointer to a pointer so I am handling it here
+	if queue.OutboundEmailAddress != nil && *queue.OutboundEmailAddress != nil {
+		updateQueue.OutboundEmailAddress = *queue.OutboundEmailAddress
+	}
+
+	// Update the queue with th new rules
+	queue, resp, err = p.routingApi.PutRoutingQueue(queueId, updateQueue)
+	if err != nil {
+		return nil, resp, fmt.Errorf("failed to update routing queue %s conditional group routing rules: %s", queueId, err)
+	}
+
+	if queue.ConditionalGroupRouting != nil && queue.ConditionalGroupRouting.Rules != nil {
+		return queue.ConditionalGroupRouting.Rules, resp, nil
+	}
```

Comment:
![Screenshot 2024-03-27 at 14 50 49 (3)](https://github.com/MyPureCloud/terraform-provider-genesyscloud/assets/108797778/14167305-64e4-48dc-9468-b8312b3bc477)
I was originally returning ConditionalGroupRouting but since all the data for this resource is in ConditionalGroupRouting.Rules and there is no other properties in ConditionalGroupRouting it was easier to just return the rules since that was all I needed

Reply:
sound